app/models/spotlight/resource.rb

```diff
@@ -100,6 +102,19 @@ def touch_exhibit!
       def write?
         Spotlight::Engine.config.writable_index
       end
+
+      def cleanup_solr_record
+        blacklight_solr.delete_by_id(document_ids, params: { softCommit: true })
+      end
+
+      def document_ids
+        document_builder.documents_to_index.to_a.map { |y| y[:id] }
```

Comment:
This will not work for a couple institutions:

- Stanford, for one, expects and allows solr documents to appear in multiple exhibits, so blindly deleting them will impact other exhibits.

- The Danes (?) have a read-only index (using ` Spotlight::Engine.config.writable_index = false`), so also wouldn't want the index blindly updated.


I'd suggest a couple alternative approaches, including:

- putting this behind a configuration setting, so institutions aren't obligated to use it
- checking for `Spotlight::SolrDocumentSidecar` instances with that document id to determine whether a document appears in multiple exhibits
- introducing a new field to the index that stores either the list of exhibits an item appears in, or a list of resources that "own" the document (perhaps `Spotlight::Engine.config.resource_global_id_field` is a relevant starting point?)